### DISCUSSION TITLE
Do users make firebase become wordpress users?

### DISCUSSION THREAD

**Discussion by gabriel-rits** [dalenguyen/firebase-wordpress-plugin#159] [General]
I have a wordpress application that works as an API for an application in Next.js.
I need to release a login on the front end of Firebase users and, after that, link those users with different types of post in wordpress.
Is it possible?

**Comment by dalenguyen** [comment]
Hi @gabriel-rits, users will become WordPress users when they login to WordPress via the plugin shortcode. If your users are created in Next.js, you can send a request and create a WordPress user, so you can link them.

It's the theory, it would be complicated in practice.

**Comment by gabriel-rits** [comment]
Hi @dalenguyen.
I believe that for now we can forget about Next.js. My question is: Using the PRo version of the plugin, will firebase users also be wordpress users? For example, can I register a post and define a firebase user as an author? Would these users be a specific type of post? Or will they not be part of the wordpress database?

**Comment by dalenguyen** [comment]
Hi @gabriel-rits, they will be normal WordPress users after login via the plugin shortcode. 

If you sign in via firebase credentials, you will be assigned as `subscriber` in WordPress. (https://wordpress.dalenguyen.me/)

You can then assign users' roles to the editor and assign posts to the users. There is no special type of post - just normal WordPress users.

**Comment by gabriel-rits** [comment]
@dalenguyen so the firebase user only becomes a wordpress user after logging in?

**Reply by dalenguyen** [reply]
Yes, this's how it works at the moment.

**Reply by gabriel-rits** [reply]
@dalenguyen is there any way for me to 'transform' all firebase users into wordpress users without performing the login action?

**Reply by dalenguyen** [reply]
You can export from firebase, then transform & import to wordpress, not sure if the password is the same or not.
